pkg/objstore/client/factory.go

```diff
@@ -18,23 +18,25 @@ import (
 var ErrNotFound = errors.New("no valid GCS or S3 configuration supplied")
 
 // NewBucket initializes and returns new object storage clients.
-// TODO(bplotka): Use that in every command.
 func NewBucket(gcsBucket *string, s3Config s3.Config, reg *prometheus.Registry, component string) (objstore.Bucket, func() error, error) {
 	if *gcsBucket != "" {
 		gcsOptions := option.WithUserAgent(fmt.Sprintf("thanos-%s/%s (%s)", component, version.Version, runtime.Version()))
 		gcsClient, err := storage.NewClient(context.Background(), gcsOptions)
 		if err != nil {
 			return nil, nil, errors.Wrap(err, "create GCS client")
 		}
-		return gcs.NewBucket(*gcsBucket, gcsClient.Bucket(*gcsBucket), reg), gcsClient.Close, nil
+		b := gcs.NewBucket(*gcsBucket, gcsClient.Bucket(*gcsBucket), reg)
+		bkt := objstore.BucketWithMetrics(*gcsBucket, b, reg)
+		return bkt, gcsClient.Close, nil
 	}
 
 	if s3Config.Validate() == nil {
 		b, err := s3.NewBucket(&s3Config, reg, component)
 		if err != nil {
 			return nil, nil, errors.Wrap(err, "create s3 client")
 		}
-		return b, func() error { return nil }, nil
+		bkt := objstore.BucketWithMetrics(s3Config.Bucket, b, reg)
```

Comment:
can we inline this as well?

Reply:
this is harder as `s3.NewBucket` returns `(*Bucket, error)` whereas `gcs.NewBucket` returns `*Bucket`

Reply:
no I rather meant here to not add unnecessary `bkt` variable.

so:
```
return objstore.BucketWithMetrics(s3Config.Bucket, b, reg), func() error { return nil }, nil
 
```